stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/TypeAdapterGenerator.java

```diff
@@ -759,6 +759,9 @@ public TypeSpec getTypeAdapterSpec(@NotNull TypeTokenConstantsGenerator typeToke
         }
 
         AnnotatedClass annotatedClass = SupportedTypesModel.getInstance().getSupportedType(typeMirror);
+        if(null == annotatedClass) {
+            throw new IllegalStateException("The AnnotatedClass class can't be null in TypeAdapterGenerator : " + typeMirror.toString());
+        }
```

Comment:
Maybe it would be better to annotate the `getSupportedType` method as `@NonNull` and throw this exception if it tries to return null in that method. What do you think?